ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/MessageEntity.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class MessageEntity extends KsqlEntity {
+
+  private final Optional<String> message;
+
+  public MessageEntity(
+      @JsonProperty("statementText")  final String statementText,
+      @JsonProperty("warnings")       final List<KsqlWarning> warnings,
+      @JsonProperty("message")        final Optional<String> message) {
+    super(statementText, warnings);
+    Objects.requireNonNull(message);
+    this.message = message;
+  }
+
+  public Optional<String> getMessage() {
+    return message;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
incorrect hashCode and equals, as they're not taking into account the parent class's fields.

I'd suggest adding a unit test using `EqualsTester` and `NullPointerTester` to ensure your class is well behaved.

Reply:
Looking at the other KSqlEntity subclasses, most (all?) seem to suffer from this problem.

So we should probably fix them all.

Reply:
I actually wonder why hashCode() and equals() are overridden for these classes in the first place.

Reply:
> So we should probably fix them all.

If they are needed, then yes, but that's a different PR ;) . What's more important is ensuring any _new_ code is correct.

> I actually wonder why hashCode() and equals() are overridden for these classes in the first place.

Totally. Often its only for tests to ensure you can assert on output.  I'm not normally a fan of having production code just for tests, but I can live with this.  However, is it's in the production code base it should be correct!

So we should either fix this or remove it.